packages/nuxt/src/app/composables/router.ts

```diff
@@ -152,7 +152,7 @@ export const navigateTo = (to: RouteLocationRaw | undefined | null, options?: Na
 
   // Early redirect on client-side
   if (import.meta.client && !isExternal && inMiddleware) {
-    return to
+    return { path: to, replace: options?.replace ?? false }
```

Comment:
`to` might be an object - would you be able to handle that? 🙏

Reply:
https://github.com/nuxt/nuxt/pull/30283/commits/3053972c4b625e0d60a184ed76c54d4949874aea
Should fix it